onecodex/viz/_distance.py

```diff
@@ -64,14 +79,18 @@ def _cluster_by_sample(
         }
 
     def _cluster_by_taxa(self, linkage=Linkage.Average):
+        import numpy as np
         from scipy.cluster import hierarchy
         from scipy.spatial.distance import squareform
         from sklearn.metrics.pairwise import euclidean_distances
 
-        dist_matrix = euclidean_distances(self._results.T).round(6)
-        clustering = hierarchy.linkage(squareform(dist_matrix), method=linkage)
+        df = self._results.dropna(how="all").replace(np.nan, 0)
+
+        dist_matrix = euclidean_distances(df.T).round(6)
```

Comment:
Could we use https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.nan_euclidean_distances.html#sklearn.metrics.pairwise.nan_euclidean_distances and not fill in 0's here?

Reply:
Scratch that, we do want the distances to 0 calculated